tests/test_file_processing.py

```diff
@@ -614,7 +630,7 @@ def test_set_state_for_file_when_in_success_folder(config):
     return False
 
 # tests for inserting docs into mlwh
-def test_insert_samples_from_docs_into_mlwh(config):
+def test_insert_samples_from_docs_into_mlwh(config, mlwh_connection):
```

Comment:
Ok so now you're clearing the table at the start of each test by doing it in the mlwh_connection method? then leaving that connection open while testing. If the test fails the finally still closes the mlwh_connection? and the tested code is opening and handling it's own separate connection. Seems neat.